server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -92,6 +94,20 @@ public BulkRequest(@Nullable String globalIndex) {
         this.globalIndex = globalIndex;
     }
 
+    /**
+     * See {@link TransportBulkAction#wrapBulkAsSingleItemResponse(ActionListener)} for wrapping translation of a bulk response
+     * to a single item response.
+     */
+    public static BulkRequest fromSingleRequest(final ReplicatedWriteRequest<?> request) {
```

Comment:
[`wrapBulkAsSingleItemResponse`](https://github.com/elastic/elasticsearch/pull/89529/files#diff-572344b1e2a22818057c05a76eecc4530d7231b347d6ff1b5f3b1686222ce5d8R178) is primarily the util method I want to port in this PR but `fromSingleRequest` (previously [`toSingleItemBulkRequest`](https://github.com/elastic/elasticsearch/blob/be7c7415627377a1b795400fb8dfcc6cbdf0e322/server/src/main/java/org/elasticsearch/action/bulk/TransportSingleItemBulkWriteAction.java#L63)) also seems useful.

I'm not sure if the intention of deprecating `TransportSingleItemBulkWriteAction` was to also deprecate `toSingleItemBulkRequest`. If that's the case I'm happy to leave it out of this PR. 

If we do want to keep it, I can also add support for `UpdateRequest` since `fromSingleRequest` currently only supports Index and Delete.

Reply:
I don't have strong opinion. Using BulkRequestBuilder is pretty easy to create bulk request with a single item, including update request. So I don't it's really necessary to add support for `UpdateRequest`. Less code to maintain is better.

Reply:
In the interest of having one way of doing things (i.e., use the request builder), I'll skip porting this method. It's got that side-effect on the passed-in request too which I'm not a fan of.